### ISSUE TITLE
Getting started: Good first issues for new contributors

### ISSUE THREAD

**Issue by Yinnerston** [Yinnerston/Teaser#29]
# Good First Issues

- [ ] Integration testing for the backend. See [`src/backend/teaser/core/tests`](../blob/master/src/backend/teaser/core/tests) for example tests.
- [ ] Add tutorials or other documentation from the codebase and [notes](https://drive.google.com/drive/folders/111fl7wlXxtrpvYRWdkBTX6MQNjATuTTx?usp=sharing)
- [ ] Bugfix get `get_general_feed_service` in [`src/backend/teaser/core/services/post_service.py`](src/backend/teaser/core/services/post_service.py). Currently it only gets posts that have been tagged, so to include all posts, a potential fix is to add an `Untagged CategoriesModel` and create a `PostCategoriesModel` foreign key relationship between an untagged post and the untagged category. Otherwise the `get_general_feed_service` can be rewritten for a NoSQL pattern.
- [ ] Define a separate feed for logged in users in `get_feed_for_you_service` in [`src/backend/teaser/core/services/post_service.py`](src/backend/teaser/core/services/post_service.py).
- [ ] Define backend endpoints to get all the liked and bookmarked posts by a user. Make a frontend screen for liked posts and bookmarked posts in the [`ProfileView`](../blob/master/src/frontend/teaser/components/templates/profile/ProfileView.js) 
- [ ] Implement some of the buttons in the [`UploadEditVideoScreen`](../blob/master/src/frontend/teaser/screens/upload/UploadEditVideoScreen.js)

**Comment by basu-10** [comment]
Hi, i am very new to github. and was on the hunt for open source contributions, specifically "good first issues," when I stumbled upon this particular one in your repo. I happen to have experience with python, and I'm eager to jump right in and contribute. Can you guide me on how to get started?

**Comment by Yinnerston** [comment]
> Hi, i am very new to github. and was on the hunt for open source contributions, specifically "good first issues," when I stumbled upon this particular one in your repo. I happen to have experience with python, and I'm eager to jump right in and contribute. Can you guide me on how to get started?

Of course, I'd love to help new contributors. The fact that you know python is good, you can work on the python backend.

### How can python developers contribute?

Firstly, you need to make sure you can run the project on your own machine. There currently is no separation of development and production environments (see issue #32 ). I'm still working on getting this project ready for new contributors to run on their own machine.  
Secondly, make sure you request to join the [Jira](https://memefeed.atlassian.net/jira/software/projects/TSR/boards/6). I'm primarily keeping track of assigning issues using Jira for now.
After you joined, assign an issue to yourself. I would recommend one of the issues above such as getting started by writing basic tests for the other backend services.
Thirdly, you need to familiarize yourself on how Django and Django Ninja work. I'd recommend reading through the django project documentation because the backend is built with Django Ninja.

Additional docs that might help:
- https://yinnerston.github.io/Teaser/docs/explaination/style_guide/#backend
- https://yinnerston.github.io/Teaser/docs/getting-started/data_model/

### Is Teaser ready for open source contributors yet?

- Honestly, no (blocked by #32 ) . 
- Currently, the frontend only has full functionality on Android. 
- I need to change the frontend code to work on a dev environment with a backend IP defined in the `.frontend.env` file.

TLDR: Still working on updating Teaser to work for open source contribution. You can assign issues to yourself but maybe check back in a week or two when #32  is completed.

**Comment by cosmoglint** [comment]
Hi @Yinnerston  I have good experience with python,django. I can help you with the local setup or with any of the other issues. let me know and I can pick it up!

**Comment by Yinnerston** [comment]
Hi @cosmoglint , I think it's awesome that you've taken an interest in Teaser.

I've written down instructions on how to setup the project locally at #32 .

I want to work on other projects and uni right now, but I'm happy to review pull requests. Check the blog in the documentation for more details~.

**Comment by matheuslfarias** [comment]
Hi @Yinnerston, I would like to consistently contribute to the project. But all the projects I developed in python and django were personal projects without much glamor. If you're in a lot of mood and patience, I'd be very happy to help!
